contracts/exchangeIssuance/FlashMintLeveragedExtended.sol

```diff
@@ -0,0 +1,590 @@
+        IWETH(addresses.weth).withdraw(amountToReturn);
+        msg.sender.transfer(amountToReturn);
+    }
+
+    function issueSetFromExactERC20(
+        ISetToken _setToken,
+        uint256 _minSetAmount,
+        address _inputToken,
+        uint256 _inputTokenAmount,
+        DEXAdapter.SwapData memory _swapDataDebtForCollateral,
+        DEXAdapter.SwapData memory _swapDataInputToken,
+        uint256 _priceEstimateInflator,
+        uint256 _maxDust
+    )
+        external
+        nonReentrant
+        returns(uint256)
+    {
+        IERC20(_inputToken).transferFrom(msg.sender, address(this), _inputTokenAmount);
+
+        return _issueSetFromExactInput(
+            _setToken,
+            _minSetAmount,
+            _inputToken,
+            _inputTokenAmount,
+            _swapDataDebtForCollateral,
+            _swapDataInputToken,
+            _priceEstimateInflator,
+            _maxDust
+        );
+    }
+
+    function issueSetFromExactETH(
+        ISetToken _setToken,
+        uint256 _minSetAmount,
+        DEXAdapter.SwapData memory _swapDataDebtForCollateral,
+        DEXAdapter.SwapData memory _swapDataInputToken,
+        uint256 _priceEstimateInflator,
+        uint256 _maxDust
+    )
+        external
+        payable
+        nonReentrant
+        returns(uint256)
+    {
+        IWETH(addresses.weth).deposit{value: msg.value}();
+        return _issueSetFromExactInput(
+            _setToken,
+            _minSetAmount,
+            DEXAdapter.ETH_ADDRESS,
+            msg.value,
+            _swapDataDebtForCollateral,
+            _swapDataInputToken,
+            _priceEstimateInflator,
+            _maxDust
+        );
+    }
+
+    function _issueSetFromExactInput(
+        ISetToken _setToken,
+        uint256 _minSetAmount,
+        address _inputToken,
+        uint256 _inputTokenAmount,
+        DEXAdapter.SwapData memory _swapDataDebtForCollateral,
+        DEXAdapter.SwapData memory _swapDataInputToken,
+        uint256 _priceEstimateInflator,
+        uint256 _maxDust
+    )
+        internal
+        returns(uint256)
+    {
+        require(_inputTokenAmount > _maxDust, "FlashMintLeveragedExtended: _inputToken must be more than _maxDust");
+
+        while (_inputTokenAmount > _maxDust) {
+            uint256 inputTokenAmountSpent = _initiateIssuanceAndReturnInputAmountSpent(
+                _setToken,
+                _minSetAmount,
+                _inputToken,
+                _inputTokenAmount,
+                _swapDataDebtForCollateral,
+                _swapDataInputToken
+            );
+            _inputTokenAmount = _inputTokenAmount - inputTokenAmountSpent;
+            uint256 priceEstimate = _minSetAmount.mul(_priceEstimateInflator).div(inputTokenAmountSpent);
+            _minSetAmount = _inputTokenAmount.mul(priceEstimate).div(1 ether);
+        }
+
+        // TODO: Decide what to do with the dust (maybe swap to eth and return as gas subsidy)
```

Comment:
i think its ideal to return dust in the form of ETH rebate here. we can also add a check that this ETH value is not too large

3 checks
- input token amount matches exactly
- set token amount is at least the minimum amount
- eth gas rebate does not exceed contract setting (if it did, that would mean misparameterization is likely)